cdap-proto/src/main/java/co/cask/cdap/proto/metadata/MetadataSearchTargetType.java

```diff
@@ -29,11 +31,28 @@
 
   private final String internalName;
 
-  private MetadataSearchTargetType(String internalName) {
+  MetadataSearchTargetType(String internalName) {
     this.internalName = internalName;
   }
 
-  public String getInternalName() {
+  public String getValue() {
     return internalName;
   }
+
+  @Override
+  public String toString() {
+    return getValue();
+  }
+
+  /**
+   * @return {@link MetadataSearchTargetType} of the given value.
+   */
+  public static MetadataSearchTargetType getEnum(String value) {
+    for (MetadataSearchTargetType metadataSearchTargetType : values()) {
+      if (metadataSearchTargetType.getValue().equalsIgnoreCase(value)) {
+        return metadataSearchTargetType;
+      }
+    }
+    throw new IllegalArgumentException("Enum not found for value: " + value);
```

Comment:
actually I think this should be ``"No enum constant for serialized form: " + value``